src/components/app/data/hooks/useHasAvailableSubsidiesOrRequests.test.js

```diff
@@ -0,0 +1,308 @@
+import useRedeemablePolicies from './useRedeemablePolicies';
+import useEnterpriseOffers from './useEnterpriseOffers';
+import { LICENSE_STATUS } from '../../../enterprise-user-subsidy/data/constants';
+
+jest.mock('./useEnterpriseCustomer');
+jest.mock('./useSubscriptions');
+jest.mock('./useBrowseAndRequest');
+jest.mock('./useCouponCodes');
+jest.mock('./useRedeemablePolicies');
+jest.mock('./useEnterpriseOffers');
+
+const appContextValue = {
+  authenticatedUser: authenticatedUserFactory(),
+};
+
+const wrapper = ({ children }) => (
+  // eslint-disable-next-line react/jsx-filename-extension
+  <AppContext.Provider value={appContextValue}>
+    {children}
+  </AppContext.Provider>
+);
+
+const mockEnterpriseCustomer = enterpriseCustomerFactory();
+
+const mockEndDateTimeOneDayOff = dayjs().add(1, 'day').toISOString();
+const mockEndDateTimeTwoDaysOff = dayjs().add(2, 'day').toISOString();
+
+describe('useHasAvailableSubsidiesOrRequests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useEnterpriseCustomer.mockReturnValue({ data: mockEnterpriseCustomer });
+    useSubscriptions.mockReturnValue({
+      data: {
+        subscriptionLicense: undefined,
+        subscriptionPlan: undefined,
+        shouldShowActivationSuccessMessage: false,
+      },
+    });
+    useBrowseAndRequest.mockReturnValue({
+      data: {
+        requests: {
+          subscriptionLicenses: [],
+          couponCodes: [],
+        },
+      },
+    });
+    useRedeemablePolicies.mockReturnValue({
+      data: {
+        redeemablePolicies: [],
+        learnerContentAssignments: null,
+      },
+    });
+    useCouponCodes.mockReturnValue({
+      data: {
+        couponCodeAssignments: [],
+      },
+    });
+    useEnterpriseOffers.mockReturnValue({ data: { enterpriseOffers: [] } });
+  });
+  it('returns an object with false and undefined values when no data is returned', () => {
+    const { result } = renderHook(() => useHasAvailableSubsidiesOrRequests(), { wrapper });
+    expect(result.current).toEqual({
+      hasActiveLicenseOrLicenseRequest: false,
+      hasAssignedCodesOrCodeRequests: false,
+      hasAvailableLearnerCreditPolicies: false,
+      hasAvailableSubsidyOrRequests: undefined,
+      learnerCreditSummaryCardData: undefined,
+    });
+  });
+  it.each([
+    {
+      mockEnterpriseOffers: { enterpriseOffers: [] },
+      mockRedeemableLearnerCreditPolicies: {
+        redeemablePolicies: [],
+        learnerContentAssignments: null,
+      },
+      mockSubscriptions: {
+        subscriptionLicense: {
+          status: LICENSE_STATUS.ACTIVATED,
+        },
+        subscriptionPlan: undefined,
+        shouldShowActivationSuccessMessage: false,
+      },
+      mockBrowseAndRequests: {
+        requests: {
+          subscriptionLicenses: [],
+          couponCodes: [],
+        },
+      },
+      mockCouponCodes: {
+        couponCodeAssignments: [],
+      },
+      mockUseHasAvailableSubsidiesOrRequestsReturn: {
```

Comment:
nit: is the `Return` at end of this property/key needed, or could it be simplified to `mockUseHasAvailableSubsidiesOrRequests`?

Edit: or actually, this is being used as expected value for the test assertion. In that case, perhaps `expectedResult` to make it more evident this object represents an expected return value, not so much a mock (which largely implies mocking an _input_).